pkg/controller/clusterdeployment/installconfigvalidation.go

```diff
@@ -45,6 +48,13 @@ func ValidateInstallConfig(cd *hivev1.ClusterDeployment, installConfig []byte) e
 		if ic.Platform.Azure.Region != cd.Spec.Platform.Azure.Region {
 			return errors.New(regionMismatchErr)
 		}
+	case platform.VSphere != nil:
+		if ic.Platform.VSphere == nil {
+			return errors.New(novSpherePlatformErr)
+		}
+		if ic.Platform.VSphere.Username == "" || ic.Platform.VSphere.Password == "" {
```

Comment:
Are we sure an empty password is never legitimate? (Not IRL of course; I'm thinking for test environments.)

Reply:
openshift-installer will reject an installconfig that has empty credentials. so this check should help improve the problem highlighted in OCPBUGSM-33627